File: src/1414-Find-the-Minimum-Number-of-Fibonacci-Numbers-Whose-Sum-Is-K/1414.py
class Solution:
    def findMinFibonacciNumbers(self, k: int) -> int:
        res, a, b = 0, 1, 1
        while b <= k:
            a, b = b, a + b

        while a:
            if a <= k:
                k -= a
                res += 1
            a, b = b - a, a
        return res

File: src/1414-Find-the-Minimum-Number-of-Fibonacci-Numbers-Whose-Sum-Is-K/1414.java
class Solution {
    public int findMinFibonacciNumbers(int k) {
        int res = 0, a = 1, b = 1, t;
        while (b <= k) {
            t = a;
            a = b;
            b += t;
        }

        while (a > 0) {
            if (a <= k) {
                k -= a;
                res++;
            }
            t = a;
            a = b - a;
            b = t;
        }
        return res;
    }
}